lemur/static/js/admin_create_and_manage_lab.js

```diff
@@ -0,0 +1,78 @@
+
+$(document).ready(function(){
+    //Modify an existing lab
+    $('.modifyLab').click(function(){
+        var labId = $(this).closest('tr').data('labid');
+        window.location.replace('/admin_modify_lab/'+labId);        
+    });
+    // The reason that we use javascript to show the modal rather than html is that this enables us
+    // to show the modal that is closest to the delete button such that the modal is within the tr
+    // that we want to delete
+    $('.deleteLab').click(function(){
+        $(this).closest('tr').find('#deleteConfirm').modal("show");
+    });
+
+    //Duplicate/Delete a lab
+    $('.duplicateLab').add($('button[name=confirm]')).click(function(){
+        var operation = $(this).data('operation');
+        var labId = $(this).closest('tr').data('labid');
+        //Communicate the name of the lab to be duplicated with python via Ajax 
+        $.ajax({
+          type: 'POST',
+          contentType: 'application/json',
+          dataType: 'json',
+          url: 'http://127.0.0.1:5000/_admin_'+operation,
+          data: JSON.stringify({'labId':labId}),
+          success: function(result){
+                    console.log(operation+' successfully');               
+                },
+          error : function(result){
+                  console.log('Fail to '+operation);
```

Comment:
I meant the failure case in particular. If you fail to make a duplicate you
want something to happen.

On Wednesday, July 27, 2016, ZiyuanZhong <notifications@github.com> wrote:

> In lemur/static/js/admin_create_and_manage_lab.js
> <https://github.com/reed-college/lemur/pull/14#discussion_r72526506>:
>
> > +    //Duplicate/Delete a lab
> > +    $('.duplicateLab').add($('button[name=confirm]')).click(function(){
> > +        var operation = $(this).data('operation');
> > +        var labId = $(this).closest('tr').data('labid');
> > +        //Communicate the name of the lab to be duplicated with python via Ajax
> > +        $.ajax({
> > +          type: 'POST',
> > +          contentType: 'application/json',
> > +          dataType: 'json',
> > +          url: 'http://127.0.0.1:5000/_admin_'+operation,
> > +          data: JSON.stringify({'labId':labId}),
> > +          success: function(result){
> > +                    console.log(operation+' successfully');
> > +                },
> > +          error : function(result){
> > +                  console.log('Fail to '+operation);
>
> The duplicated lab will be directly shown on the webpage.
> In the following example, I click the duplicate button in order to
> duplicate 2Cortisol. A copy is created and shown on the page after I click
> on the button.
> [image: screen shot 2016-07-27 at 2 21 27 pm]
> <https://cloud.githubusercontent.com/assets/12670254/17192915/7796ac68-5405-11e6-8b8a-c8499fc51ab8.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/reed-college/lemur/pull/14/files/b1d2aba171ed0923e7c3d00bd3ea6bd2575530f1#r72526506>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgCe6Sg76PdH_ngq6TpbSdWOXfzhM7Uks5qZ8wigaJpZM4JRWdG>
> .
>


-- 
Andrew Winterman
714 362 6823